src/GitTfs/Core/GitTfsRemote.cs

```diff
@@ -386,9 +396,9 @@ public IFetchResult FetchWithMerge(int mergeChangesetId, bool stopOnFailMergeCom
                         return fetchResult;
                     }
                     var parentSha = (renameResult != null && renameResult.IsProcessingRenameChangeset) ? renameResult.LastParentCommitBeforeRename : MaxCommitHash;
-                    var isFirstCommitInRepository = (parentSha == null);
+                    var isFirstTFSCommitInRepository = (MaxChangesetId == 0);
```

Comment:
Obviously, this is the "essence" of this change.

Reply:
Yes. Everything else is just testing and comments.